packages/expo-doctor/src/doctor.ts

```diff
@@ -0,0 +1,103 @@
+import { getConfig } from '@expo/config';
+import chalk from 'chalk';
+
+import { ExpoConfigSchemaCheck } from './checks/ExpoConfigSchemaCheck';
+import { GlobalPrereqsVersionCheck } from './checks/GlobalPrereqsVersionCheck';
+import { IllegalPackageCheck } from './checks/IllegalPackageCheck';
+import { InstalledDependencyVersionCheck } from './checks/InstalledDependencyVersionCheck';
+import { SupportPackageVersionCheck } from './checks/SupportPackageVersionCheck';
+import { DoctorCheck, DoctorCheckParams } from './checks/checks.types';
+import { gteSdkVersion } from './utils/gteSdkVersion';
+import { logNewSection } from './utils/ora';
+import { warnUponCmdExe } from './warnings/windows';
+
+export async function runCheckAsync(
+  check: DoctorCheck,
+  checkParams: DoctorCheckParams
+): Promise<boolean> {
+  try {
+    const ora = logNewSection(check.description);
+    const result = await check.runAsync(checkParams);
+    if (!result.isSuccessful) {
+      ora.fail();
+      if (result.issues.length) {
+        console.log(chalk.underline(chalk.yellow(`Issues:`)));
+        console.group();
+        for (const issue of result.issues) {
+          console.log(chalk.yellow(`${issue}`));
+        }
+        console.groupEnd();
+      }
+      if (result.advice.length) {
+        console.log(chalk.underline(chalk.green(`Advice:`)));
+        console.group();
+        for (const advice of result.advice) {
+          console.log(chalk.green(`• ${advice}`));
+        }
+        console.groupEnd();
+      }
+      return false;
+    } else {
+      ora.succeed();
+      return true;
+    }
+  } catch (e: any) {
+    if (e.code === 'ENOTFOUND') {
+      console.log(
+        chalk.red(
+          `Error: this check requires a connection to the Expo API. Please check your network connection.`
+        )
+      );
+    } else {
+      console.log(chalk.red(`Error: check could not be completed due to an unexpected error.`));
+    }
+    return false;
+  }
+}
+
+export async function actionAsync(projectRoot: string) {
+  await warnUponCmdExe();
+
+  const { exp, pkg } = getConfig(projectRoot);
+
+  // expo-doctor relies on versioned CLI, which is only available for 44+
```

Comment:
I thought versioned CLI was only officially supported from SDK 46 onwards (included as beta in SDK 45)?

Reply:
you are correct! (no idea where I got the idea otherwise...)